### ISSUE TITLE
Streaming radio in MPD is laggy

### ISSUE THREAD

**Issue by AdmiralStipe** [Poeschl/Hassio-Addons#104]
**Which addon?**
MPD

**Describe the bug**
Playing of local (mp3) files works OK, but playing any online radio stream is laggy - it plays OK for a couple of seconds (or even minutes), but then jumps forward for a half of second or so and repeats to do it randomly. It only happens inside HA, if I open the stream on Raspbian in VLC, it plays smoothly. Internet connection is not an issue (30 Mbps).
Maybe also worth mentioning (although most likely not related), that volume control is not working - it always remains on zero, even if I move it higher, it jumps back to zero. Nevertheless, music is playing as MPD volume has no relation to the volume on Raspberry, but the error (exception: Failed to open mixer for 'Hassio Audio': no such mixer control: PCM) keeps popping out in log on each stream play. 

**To Reproduce**
Add the stream (f.e. http://stream.rockradio.si:9034/;stream/1) to the playlist and play it.

**Expected behavior**
Smooth (buffered) play

**System**
Operating system
Raspbian GNU/Linux 10 (buster) on Raspberry PI 4B
Docker version
19.03.13
* Supervisor version: 247
* Host system version: Home Assistant 0.116.2, running on Raspberry Pi 4B in Docker (Supervised install)

**Comment by Poeschl** [comment]
Seems that is also an issue of the alsa-pulseaudio layer. Im planing on switching the audio-backend to pulseaudio. This will also fix the volume slider.

Unfortunately I wasn't able so far.

**Comment by Poeschl** [comment]
Can you check this issue against the latest version of the `mpd` addon?

I switched the audio output to the recommended Pulseaudio output.

**Comment by AdmiralStipe** [comment]
I would love to try it, but HA does not offer me any updates. MPD still shows version 1.3.2 and no updates available (despite restarting and refreshing). Btw, I installed it through supervisor - is there a way to install/update it manually?

**Comment by Poeschl** [comment]
@AdmiralStipe On the Addon Store, in the menu on the top right is a "Releoad Addons" button. This should refresh your addon list

**Comment by AdmiralStipe** [comment]
Yeah, thx for the "reload" tip, it worked :). 
And at the same time after a few minutes of listening I'm glad to report that (at least on first sight) streaming now seems smooth, but I will need some more time to test it long-term. 
Additionally, volume slider is working now too :).
 
Thx a lot for your effort and promptness, great support.

**Comment by Poeschl** [comment]
Switching to the recommended sound output was already on my list, but now there was time for that ;) 

I would suggest you close the issue, when no further issues are noticeable.

**Comment by AdmiralStipe** [comment]
After extensive testing of different streams unfortunately I have to report, that the lag is still present - partly the same way as already reported (skipping the stream for short period of time) but now additionally also breaking the stream (short silence periods, estimated about 50ms). Truth to be told, it doesn't happen as frequently as before, but it still does. Before the change it was kind of expected (happening roughly at expected intervals) but now it's more random (happens many times in a short period of time, but then it plays smoothly for minutes). Also I have noticed that these interruptions happen more frequently at the beggining of stream (f.e. when I change the radio station) but after some time they become less frequent. Buffer related maybe? Is there any possibility to influence the buffer size or other settings?
Playing of local mp3 files still works perfectly.

**Comment by Poeschl** [comment]
I'm sorry to hear that. Hm, after looking at the mpd docs I found a [audio_buffer_size](https://www.musicpd.org/doc/html/user.html?highlight=audio_buffer_size#buffer-settings) setting.
That could be the culprit here.
Can you try running mpd with an custom mpd.conf and an bigger buffer? Instructions should in Docs.md.
I'm not able to reproduce it on my system, for unknown reasons.

**Comment by AdmiralStipe** [comment]
I had some trouble finding default mpd.conf in Git (location of the file has changed in the last version :) ), but eventually I found it and changed the buffer from default 4MB to 8MB. Unfortunately the problem persists, so it most likely isn't buffer related after all. 
Or, maybe my synthax wasn't correct and therefore had no influence... I put it this way: 

# An example of a pulseaudio output (streaming to a remote pulseaudio server)
#
audio_buffer_size		"8"
audio_output {
	type		"pulse"
	name		"Home Assistant Pulseaudio"
#	server		"remote_server"		# optional
#	sink		"remote_server_sink"	# optional
}
#

Do you think this is correct (instructions in the link you posted are not 100% clear to me)? This synthax was the only one which let me run the plugin. When I tried other synthaxes (adding MB to the number, not using quotation-marks or similar) the plugin became unavailable after restart. 
Btw, some other MPD configs (found by googling) use kB instead MB for audio_buffer_size, but i don't dare trying to change number 8 to 8092 :).

**Comment by Poeschl** [comment]
The setting is not set by default, so I created a test-config for you ;) [mpd.conf.zip](https://github.com/Poeschl/Hassio-Addons/files/5428864/mpd.conf.zip)

Place the file at `/share/mpd/mpd.conf` on your HA installation and add the line `custom_config: /share/mpd/mpd.conf` to the addon config. It should load the custom configuration with the custom buffer size. If working, I will include the default of 4MB buffer as default.

**Comment by AdmiralStipe** [comment]
Thx for your effort, but the file didn't work at first, I had to tweak it a bit. Namely, the definition of buffer is probably not in correct synthax, as my MPD didn't even load upon restart when using this file directly (player not available). 
So I changed the buffer line in a way, that I added quotation marks to the number (as all other definitions in the file have it). 

audio_buffer_size "4096"

After that the player loaded normaly, unfortunately this new setting didn't have much effect on the issue. I also tried doubling the number to 8192, but the issue remained - maybe a little less frequent, but still present. 

I don't want to bother you anymore and waste wour time with this as I am more and more assuming, that the problem may be on my side (something related to my general HA or Raspberry configuration). So I will close this issue and try exploring further locally.
